### PR TITLE
turret/vehicle ammo fix based on research

### PR DESCRIPTION
**Pull request by misterprimus**
Kill and loot a turret/robot.
#### Summary

```SUMMARY: Balance "Make robot/turret ammo match research."```
#### Purpose of change

To allow chickenwalker/tank to drop ammo and to match to research how much ammo the automated vehicles and turrets would carry.
#### Describe the solution

N/A - not really a feature, which was implemented in an earlier PR. This is just JSON adjustments.

I looked up how much these carried, but only came up with the M240 weapon, which is stated to carry 450 rounds of 7.62x51. I found the weight of all the cartridges used by the robots in the game and made the TOTAL ammo weight to be about the same (rounding the ammo qty to an even number).
image

#### Testing

Spawned and destroyed chickenwalker.
#### Additional context

Got rid of M16/M4 and replaced with M249 to prevent spawning of mags and make it look like the gun was being operated using a belt, which appears to be the way they are operated IRL.

### CODE REVIEW

File: data/json/monsters/turrets.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "armor_cut": 16,
     "vision_night": 3,
     "revert_to_itype": "bot_turret",
-    "starting_ammo": { "9mm": 30 },
+    "starting_ammo": { "9mm": 1500 },
```
Comment (Rail-Runner) [inline data/json/monsters/turrets.json]: This should stay 30, I think; this is a turret improvised by player which uses a normal MP5 magazine, which is 30 rounds.

Comment (misterprimus) [reply data/json/monsters/turrets.json]: Ah yes didn't realize it was an improvised turret. Which turret is the one that appears in science stations and fires 9mm?

Comment (anothersimulacrum) [reply data/json/monsters/turrets.json]: There is none.

Comment (misterprimus) [reply data/json/monsters/turrets.json]: Really? The 9mm 100 round science lab turrets are gone? Were they ever replaced with anything similar or is it just zeds now?

Comment (anothersimulacrum) [reply data/json/monsters/turrets.json]: They were obsoleted and replaced with CROWS.
They were then un-obsoleted to make these survivor 9mm turrets.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -1140,6 +1140,23 @@
...
     "flags": [ "TRADER_AVOID", "NO_REPAIR" ]
   },
+  {
+    "type": "GENERIC",
+    "id": "broken_chickenbot",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken chickenbot",
+    "category": "other",
+    "description": "A broken chickenwalker.  Still looks intimidating despite being permanently inoperative, possibly due to the sheer size and mass.  Could be gutted for parts.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]:
```suggestion
    "description": "A broken chicken walker.  Still looks intimidating despite being permanently inoperative, possibly due to the sheer size and mass.  Could be gutted for parts.",
```

File: data/json/recipes/recipe_deconstruction.json [hunk 3]
```diff
@@ -947,6 +947,30 @@
...
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "memory_module", 1 ] ],
+      [ [ "pathfinding_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "chickenbot_chassis", 1 ] ],
+      [ [ "targeting_module", 1 ] ],
+      [ [ "gun_module", 3 ] ],
+      [ [ "m4a1", 1 ] ],
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_deconstruction.json]: Should chickenbot be equipped with M4?

```suggestion
      [ [ "m249", 1 ] ],
```

**Comment by misterprimus** [comment]
> `mon_secubot` having an M16A4 is completely deliberate. Add a new `mon_talon_m249` if you want but don't touch `mon_secubot`.
> 
> Chicken walkers are on the way to being obsolete. See #35821.

The reason I did that was to avoid having the TALON drop STANAG mags. How would that thing reload its gun? Pretty sure they are belt-fed, right?

**Comment by ymber** [comment]
The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.

**Comment by misterprimus** [comment]
> The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.

You mean like when they spawn in banks? OK then maybe I will add an m249 for places like labs and outposts.

**Comment by misterprimus** [comment]
> The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.

Has there ever been a 30-round mag-fed SWORDs unit IRL? I read they can use M16 rifles, but with a 30 round mag?

**Comment by tenmillimaster** [comment]
> > The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.
> 
> Has there ever been a 30-round mag-fed SWORDs unit IRL? I read they can use M16 rifles, but with a 30 round mag? Just doesn't follow the script if they are giving it hundreds of rounds of a bigger cartridge.

Yes, they're not intended to function without an operator.

**Comment by misterprimus** [comment]
> > > The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.
> > 
> > 
> > Has there ever been a 30-round mag-fed SWORDs unit IRL? I read they can use M16 rifles, but with a 30 round mag? Just doesn't follow the script if they are giving it hundreds of rounds of a bigger cartridge.
> 
> Yes, they're not intended to function without an operator.

Seems like even a bank secubot would be expected to function without an operator during a shootout situation, though. And it would likely require potentially more than 30 rounds if they think security is a big enough risk to get armed guards. Labs would be even more dangerous since secubots there would be intended to deal with zed groups and high level monsters breaking out.

**Comment by misterprimus** [comment]
> The robot can't reload on its own. The operator has to reload it which is why belt fed weapons are much more common on UGVs. I gave this one an M16A4 because it's much more likely that the places these spawn would have M16s than M249s and they're not for applications where you're likely to need more than 30 rounds.

Anyway I guess I'll just leave it as is with the m16 w/ 30 rounds, though it doesn't make sense to me in many of the settings (a box mag would make more sense).

**Comment by misterprimus** [comment]
> The remaining changes in this look alright.

Yes maybe I'll add a proper "military grade" TALON (with M249) in another PR.

**Comment by ymber** [comment]
This hasn't changed since I last reviewed it. I don't think it's particularly useful given that chicken walkers and tank drones are on the way out but I don't specifically object to anything in here.

**Comment by misterprimus** [comment]
> This hasn't changed since I last reviewed it. I don't think it's particularly useful given that chicken walkers and tank drones are on the way out but I don't specifically object to anything in here.

I clicked the button since your last post indicated you were OK with the changes, but the reviewer changes requested was still up. Thought I needed to click that to move things along.

While not exactly shockingly useful, it will make the walker be consistent with the beagle and other robots and seems more realistic. That PR you linked says "Removes chicken walkers and beagle minitanks from military roadblocks and from any spawn lists (edit to clarify, they are **still specifically called up in some labs and in the necropolis**)." No clue what the final verdict is or will be with that PR or when (if ever) it will be merged. No point in having a realism/consistency hole in the game just because we're waiting for a PR that might not get merged and which doesn't even necessarily conflict with this change.

File: data/json/items/corpses/inactive_bots.json [hunk 4]
```diff
@@ -696,6 +696,33 @@
       "skill2": "computer"
     }
   },
+  {
+    "id": "broken_chickenbot",
+    "looks_like": "broken_tripod",
+    "type": "TOOL",
+    "name": "inactive chicken walker",
+    "description": "This is an inactive chicken walker. Using this item involves placing it on the ground, loading the unit with the factory-loaded 5.56 rounds and 40mm cartridge grenades in your inventory (if you wish to divide your ammunition, set aside whatever ammunition you do NOT want to give the robot) and turning it on.  If reprogrammed and rewired successfully the tank drone will then identify you as a friendly, roam around or follow you, and attack all enemies with a built-in firearm and grenade launcher.",
```
Comment (ZhilkinSerg) [inline data/json/items/corpses/inactive_bots.json]:
```suggestion
    "description": "This is an inactive chicken walker.  Using this item involves placing it on the ground, loading the unit with the factory-loaded 5.56 rounds and 40mm cartridge grenades in your inventory (if you wish to divide your ammunition, set aside whatever ammunition you do NOT want to give the robot) and turning it on.  If reprogrammed and rewired successfully the tank drone will then identify you as a friendly, roam around or follow you, and attack all enemies with a built-in firearm and grenade launcher.",
```

File: data/json/items/corpses/inactive_bots.json [hunk 5]
```diff
@@ -696,6 +696,33 @@
...
+    "to_hit": -3,
+    "bashing": 8,
+    "material": [ "steel" ],
+    "symbol": ";",
+    "color": "light_green",
+    "use_action": {
+      "type": "place_monster",
+      "monster_id": "mon_chickenbot",
+      "friendly_msg": "The chicken bot rolls out and begins acquiring targets.",
+      "hostile_msg": "The chicken bot swivels its turret and aims directly at you. Don your brown pants!",
```
Comment (ZhilkinSerg) [inline data/json/items/corpses/inactive_bots.json]:
```suggestion
      "hostile_msg": "The chicken bot swivels its turret and aims directly at you.  Don your brown pants!",
```

File: data/json/items/corpses/inactive_bots.json [hunk 6]
```diff
@@ -696,6 +696,33 @@
...
+      "//": "Milspec, clearly designed with little concern for collateral damage. What did you expect of a chicken walker?",
```
Comment (ZhilkinSerg) [inline data/json/items/corpses/inactive_bots.json]:
```suggestion
      "//": "Milspec, clearly designed with little concern for collateral damage.  What did you expect of a chicken walker?",
```

**Pull request merged**